### PR TITLE
Move grad_norm to a dedicated utilities file 

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

This PR further trims dependencies from the core LightningModule by turning `grad_norm` into a function that operates on an nn.Module instead of being defined on a LightningModule via this mixin.

Similar in spirit to #7180 and #6834 to simplify inheritance

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by kaushikb11 — APPROVED** [review]
LGTM!

File: pytorch_lightning/core/grads.py [hunk 1]
```diff
@@ -36,20 +38,12 @@ def grad_norm(self, norm_type: Union[float, int, str]) -> Dict[str, float]:
...
-            norms[f'grad_{norm_type}_norm_{name}'] = round(param_norm, 4)
 
-            all_norms.append(param_norm)
-
-        total_norm = float(torch.tensor(all_norms).norm(norm_type))
-        norms[f'grad_{norm_type}_norm_total'] = round(total_norm, 4)
-
-        return norms
+        .. deprecated::v1.3
+            Will be removed in v1.5.0. Use ``pytorch_lightning.utilities.grads.grad_norm`` instead.
```
Comment (awaelchli) [inline pytorch_lightning/core/grads.py:43]:
```suggestion
            Will be removed in v1.5.0. Use :func:`pytorch_lightning.utilities.grads.grad_norm` instead.
```

File: pytorch_lightning/core/grads.py [hunk 2]
```diff
@@ -36,20 +38,12 @@ def grad_norm(self, norm_type: Union[float, int, str]) -> Dict[str, float]:
...
+        """
+        rank_zero_deprecation(
+            "LightningModule.grad_norm is deprecated in v1.3 and will be removed in v1.5."
+            "Use grad_norm from pytorch_lightning.utilities.grads instead."
```
Comment (awaelchli) [inline pytorch_lightning/core/grads.py:47]:
```suggestion
            "LightningModule.grad_norm is deprecated in v1.3 and will be removed in v1.5."
            " Use grad_norm from pytorch_lightning.utilities.grads instead."
```

File: pytorch_lightning/core/grads.py [hunk 3]
```diff
@@ -36,20 +38,12 @@ def grad_norm(self, norm_type: Union[float, int, str]) -> Dict[str, float]:
             norms: The dictionary of p-norms of each parameter's gradient and
                 a special entry for the total p-norm of the gradients viewed
                 as a single vector.
-        """
```
Comment (awaelchli) [inline pytorch_lightning/core/grads.py:39]: do we usually remove the docstrings too? there is a small chance they would go out of sync between 1.3 and 1.5 who knows.

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/core/grads.py [hunk 4]
```diff
@@ -36,20 +38,12 @@ def grad_norm(self, norm_type: Union[float, int, str]) -> Dict[str, float]:
...
-            param_norm = float(p.grad.data.norm(norm_type))
-            norms[f'grad_{norm_type}_norm_{name}'] = round(param_norm, 4)
 
-            all_norms.append(param_norm)
-
-        total_norm = float(torch.tensor(all_norms).norm(norm_type))
-        norms[f'grad_{norm_type}_norm_total'] = round(total_norm, 4)
-
-        return norms
+        .. deprecated::v1.3
```
Comment (justusschock) [inline pytorch_lightning/core/grads.py:42]:
```suggestion
        .. deprecated:: v1.3
```
I think without the space sphinx doesn't render it properly

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/core/grads.py (hunk 3 above)
Comment (carmocca) [reply pytorch_lightning/core/grads.py:39]: Yes, let's remove this docstring

File: pytorch_lightning/utilities/grads.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        norm_type: The type of the used p-norm, cast to float if necessary.
+            Can be ``'inf'`` for infinity norm.
+
+    Return:
+        norms: The dictionary of p-norms of each parameter's gradient and
+            a special entry for the total p-norm of the gradients viewed
+            as a single vector.
+
+    .. deprecated::v1.3
+        Will be removed in v1.5.0.
```
Comment (carmocca) [inline pytorch_lightning/utilities/grads.py:38]: This is not deprecated...

```suggestion
```

**Review by carmocca — APPROVED** [review]
